WalletWasabi.Fluent/ViewModels/CoinJoinProfiles/CoinJoinProfilesViewModel.cs

```diff
@@ -75,22 +75,26 @@ private async Task OnManualSetupAsync()
 		}
 	}
 
-	private void OnNext(KeyManager keyManager, bool isNewWallet)
+	private void OnNext(IWalletModel walletModel, bool isNewWallet)
 	{
 		var selected = SelectedProfile ?? SelectedManualProfile ?? Profiles.First();
 
-		keyManager.RedCoinIsolation = selected.RedCoinIsolation;
-		keyManager.SetAnonScoreTarget(selected.AnonScoreTarget, toFile: false);
-		keyManager.SetFeeRateMedianTimeFrame(selected.FeeRateMedianTimeFrameHours, toFile: false);
-		keyManager.IsCoinjoinProfileSelected = true;
+		using (walletModel.Settings.BatchChanges())
+		{
+			walletModel.Settings.RedCoinIsolation = selected.RedCoinIsolation;
+			walletModel.Settings.AnonScoreTarget = selected.AnonScoreTarget;
+			walletModel.Settings.FeeRateMedianTimeFrameHours = selected.FeeRateMedianTimeFrameHours;
+			walletModel.Settings.IsCoinjoinProfileSelected = true;
+		}
 
 		if (isNewWallet)
 		{
-			Navigate().To().AddedWalletPage(keyManager);
+			// TODO: remove this after AddedWalletPage is decoupled
+			var wallet = Services.WalletManager.GetWallets(false).First(x => x.WalletName == walletModel.Name);
```

Comment:
I doubt this works. `isNewWallet` is true, which means the wallet doesn't exist yet.

Reply:
Yepp, just tested it:

```
2023-06-06 21:02:27.270 [1] ERROR       Program.RunAsGuiAsync (164)     System.InvalidOperationExcep
tion: Sequence contains no matching element
   at System.Linq.ThrowHelper.ThrowNoMatchException()
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at WalletWasabi.Fluent.ViewModels.CoinJoinProfiles.CoinJoinProfilesViewModel.OnNext(IWalletModel
walletModel, Boolean isNewWallet) in WalletWasabi.Fluent\ViewModels\CoinJoinProfiles\CoinJoinProfile
sViewModel.cs:line 93
   at WalletWasabi.Fluent.ViewModels.CoinJoinProfiles.CoinJoinProfilesViewModel.<>c__DisplayClass1_0
.<.ctor>b__0() in WalletWasabi.Fluent\ViewModels\CoinJoinProfiles\CoinJoinProfilesViewModel.cs:line
21
   at ReactiveUI.ReactiveCommand.<>c__DisplayClass0_0.<Create>b__1(IObserver`1 observer) in D:\a\Rea
ctiveUI\ReactiveUI\src\ReactiveUI\ReactiveCommand\ReactiveCommand.cs:line 109
   at System.Reactive.Linq.QueryLanguage.CreateWithDisposableObservable`1.SubscribeCore(IObserver`1
observer) in /_/Rx.NET/Source/src/System.Reactive/Linq/QueryLanguage.Creation.cs:line 35
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer) in /_/Rx.NET/Source/src/Syste
m.Reactive/ObservableBase.cs:line 58
```

Try to create a new wallet to reproduce the exception.

Reply:
@soosr fixed in latest commit. I introduced a temporary fix since the final version of this is in #10730